### ISSUE TITLE
BUG: zaya_forward_batch silently truncates B to 8 with no error

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#264]
## Summary

`src/zaya_engine.cpp`'s `zaya_forward_batch()` silently caps the batch size to 8 with no warning:

```cpp
if (B > 8) B = 8; // safety cap
```

If a caller passes B=16 expecting all 16 tokens to be processed, only the first 8 get computed and tokens 9-16 are silently dropped. No log, no error code, no assertion.

## Impact

- Data loss in batched inference: half the prompt tokens don't get processed
- Hard to debug because the function returns void and the caller has no way to know
- The B=8 limit itself is reasonable (constrained by shared memory), but the silent truncation is not

## Location

`src/zaya_engine.cpp` line ~496:

```cpp
void zaya_forward_batch(ZayaState* s, const int* token_ids, float* logits_out, int B) {
    int g1 = (eng.h+BLK-1)/BLK;
    if (B > 8) B = 8; // safety cap
```

## Fix

Replace silent truncation with a logged warning:

```cpp
if (B > 8) {
    fprintf(stderr, "zaya_forward_batch: B=%d > 8, truncating to 8 (tokens %d-%d will NOT be processed)\n", B, 8, B-1);
    B = 8;
}
```

Or better, return an error code so the caller can handle it appropriately.

## Bonus

The comment says "safety cap" but 8 is the max batch size supported by the batch-union MoE kernels. This should be a named constant (`MAX_BATCH = 8`) at the top of the file, not an inline magic number.
